examples/step-89/step-89.cc

```diff
@@ -0,0 +1,1884 @@
+
+  // Setup MatrixFree.
+
+  pcout << "Create DoFHandler..." << std::endl;
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(FESystem<dim>(FE_DGQ<dim>(degree), dim + 1));
+  pcout << " - Number of DoFs: " << dof_handler.n_dofs() << std::endl;
+
+  AffineConstraints<Number> constraints;
+  constraints.close();
+
+  pcout << "Setup MatrixFree..." << std::endl;
+  typename MatrixFree<dim, Number>::AdditionalData data;
+  data.mapping_update_flags = update_gradients | update_values;
+  data.mapping_update_flags_inner_faces =
+    update_quadrature_points | update_values;
+  data.mapping_update_flags_boundary_faces =
+    data.mapping_update_flags_inner_faces;
+
+  MatrixFree<dim, Number> matrix_free;
+  matrix_free.reinit(
+    MappingQ1<dim>(), dof_handler, constraints, QGauss<dim>(degree + 1), data);
+
+
+  pcout << "Run vibrating membrane testcase..." << std::endl;
+  // Vibrating membrane testcase:
+  //
+  // Homogenous pressure DBCs are applied for simplicity. Therefore,
+  // modes can not be chosen arbitrarily.
+  const double                                            modes = 10.0;
+  std::map<types::material_id, std::pair<double, double>> homogenous_material;
+  homogenous_material[numbers::invalid_material_id] = std::make_pair(1.0, 1.0);
+  const auto initial_solution_membrane =
+    Step89::InitialConditionVibratingMembrane<dim>(modes);
+
+  pcout << " - Point-to-point interpolation: " << std::endl;
+  // Run vibrating membrane testcase using point-to-point interpolation:
+
+  Step89::run_with_point_to_point_interpolation(
+    matrix_free,
+    non_matching_faces,
+    homogenous_material,
+    2.0 * initial_solution_membrane.get_period_duration(
+            homogenous_material.begin()->second.first),
+    initial_solution_membrane,
+    "vm-p2p");
+
+  pcout << " - Nitsche-type mortaring: " << std::endl;
+  // Run vibrating membrane testcase using Nitsche-type mortaring:
+  Step89::run_with_nitsche_type_mortaring(
+    matrix_free,
+    non_matching_faces,
+    homogenous_material,
+    2.0 * initial_solution_membrane.get_period_duration(
+            homogenous_material.begin()->second.first),
+    initial_solution_membrane,
+    "vm-nitsche");
+
+  pcout << "Run testcase with inhomogenous material..." << std::endl;
+  // In-homogenous material testcase:
+  //
+  // Run simple testcase with in-homogenous material and Nitsche-type mortaring:
+  std::map<types::material_id, std::pair<double, double>> inhomogenous_material;
+  inhomogenous_material[0] = std::make_pair(1.0, 1.0);
+  inhomogenous_material[1] = std::make_pair(3.0, 1.0);
+  Step89::run_with_nitsche_type_mortaring(matrix_free,
+                                          non_matching_faces,
+                                          inhomogenous_material,
+                                          /*runtime*/ 0.3,
+                                          Step89::GaussPulse<dim>(0.3, 0.5),
+                                          "inhomogenous");
```

Comment:
I am wondering if we could guard these lines and only run them if deal.II is compiled with CGAL?

Reply:
Generally, there are no comments that CGAL is used for creating the intersections.